### PR TITLE
Add `custom-folder-icon` to the managed set

### PR DESCRIPTION
**Pull request by strangelookingnerd**

### CODE REVIEW

File: bom-weekly/pom.xml [hunk 1]
```diff
@@ -299,6 +299,11 @@
                 <artifactId>credentials-binding</artifactId>
                 <version>523.vd859a_4b_122e6</version>
             </dependency>
+            <dependency>
+                <groupId>org.jenkins-ci.plugins</groupId>
```
Comment (basil) [inline bom-weekly/pom.xml:303]: Incorrect group ID:

```suggestion
                <groupId>io.jenkins.plugins</groupId>
```

After applying this suggestion, please test locally with `PLUGINS=custom-folder-icon bash local-test.sh` and apply any other necessary fixes.

Comment (strangelookingnerd) [reply bom-weekly/pom.xml]: Seems like the heat yesterday was really getting into my head if even c&p fails me 🥵 Fixed it now. 
I ran the tests locally without any further issues.

Comment (jetersen) [reply bom-weekly/pom.xml]: I hear ya 🥵

We are all human! Thanks for contributing to Jenkins and BOM 👏

**Review by jetersen — APPROVED** [review]

**Review by basil — COMMENTED** [review]
The latest CI build failed on the `bom-2.346` with:

```
Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:3.32:validate-hpi (default-validate-hpi) on project sample: Dependency io.jenkins.plugins:custom-folder-icon:jar:2.0 requires Jenkins 2.357 or higher. -> [Help 1]
```

The way this works is we have the following BOM lines:

```
bom-weekly/pom.xml
bom-2.361.x/pom.xml
bom-2.346.x/pom.xml
bom-2.332.x/pom.xml
bom-2.319.x/pom.xml
```

Each line inherits from the preceding line. So `bom-2.361` inherits from `bom-weekly` and overrides (to older versions) any plugins where the latest version does not run on the 2.361.x LTS line.

Since `custom-folder-icon` 2.0 requires Jenkins 2.357 or higher, version 2.0 will work for the `bom-weekly` line and the `bom-2.361` line, but it won't work on the `bom-2.346` or older lines. So you'll need to at least edit `bom-2.346` to provide an older version for that line and below.

**Comment by strangelookingnerd** [comment]
@basil Thanks for the clarification. Fixed in my latest commit.

Yet the build still fails, not sure if this is an issue on github's end?

```
[2022-08-17T07:51:12.525Z] [INFO] --------------< org.jenkins-ci.plugins:cloudbees-folder >---------------
[2022-08-17T07:51:12.525Z] [INFO] Building Folders Plugin 6.999999-SNAPSHOT
[2022-08-17T07:51:12.525Z] [INFO] --------------------------------[ hpi ]---------------------------------
[2022-08-17T07:51:13.935Z] Error : The git-clone command failed. || Cloning into '/home/jenkins/workspace/Tools_bom_PR-1406/pct-work/custom-folder-icon'...
[2022-08-17T07:51:13.935Z] ssh: connect to host github.com port 22: Connection timed out
[2022-08-17T07:51:13.935Z] fatal: Could not read from remote repository.
[2022-08-17T07:51:13.935Z] 
[2022-08-17T07:51:13.935Z] Please make sure you have the correct access rights
[2022-08-17T07:51:13.935Z] and the repository exists.
[2022-08-17T07:51:13.935Z] 
[2022-08-17T07:51:13.935Z] Aug 17, 2022 7:51:00 AM org.jenkins.tools.test.PluginCompatTester testPlugins
[2022-08-17T07:51:13.935Z] SEVERE: Internal error while executing a test for core 2.319.3 and plugin custom-folder-icon 1.11. Please submit a bug to plugin-compat-tester
[2022-08-17T07:51:13.935Z] org.jenkins.tools.test.exception.PluginSourcesUnavailableException: Problem while checking out plugin sources!
[2022-08-17T07:51:13.935Z] 	at org.jenkins.tools.test.PluginCompatTester.testPluginAgainst(PluginCompatTester.java:524)
[2022-08-17T07:51:13.935Z] 	at org.jenkins.tools.test.PluginCompatTester.testPlugins(PluginCompatTester.java:328)
[2022-08-17T07:51:13.935Z] 	at org.jenkins.tools.test.PluginCompatTesterCli.main(PluginCompatTesterCli.java:191)
[2022-08-17T07:51:13.935Z] Caused by: java.lang.RuntimeException: The git-clone command failed. || Cloning into '/home/jenkins/workspace/Tools_bom_PR-1406/pct-work/custom-folder-icon'...
[2022-08-17T07:51:13.935Z] ssh: connect to host github.com port 22: Connection timed out
[2022-08-17T07:51:13.935Z] fatal: Could not read from remote repository.
[2022-08-17T07:51:13.935Z] 
[2022-08-17T07:51:13.935Z] Please make sure you have the correct access rights
[2022-08-17T07:51:13.935Z] and the repository exists.
[2022-08-17T07:51:13.935Z] 
[2022-08-17T07:51:13.935Z] 	at org.jenkins.tools.test.PluginCompatTester.cloneFromSCM(PluginCompatTester.java:648)
[2022-08-17T07:51:13.935Z] 	at org.jenkins.tools.test.PluginCompatTester.testPluginAgainst(PluginCompatTester.java:507)
[2022-08-17T07:51:13.935Z] 	... 2 more
[2022-08-17T07:51:13.935Z] 
```

... which is later on causing ...

```
[2022-08-17T07:51:13.936Z] + grep -q -F -e '<status>INTERNAL_ERROR</status>' pct-report.xml
[2022-08-17T07:51:13.936Z] + echo 'PCT failed with internal error'
[2022-08-17T07:51:13.936Z] PCT failed with internal error
[2022-08-17T07:51:13.936Z] + cat pct-report.xml
[2022-08-17T07:51:13.936Z] <?xml version="1.0" encoding="ISO-8859-1"?>
[2022-08-17T07:51:13.936Z] <?xml-stylesheet href="pct-report.xsl" type="text/xsl"?>
[2022-08-17T07:51:13.936Z] <report>
[2022-08-17T07:51:13.936Z]   <pluginCompatTests class="tree-map">
[2022-08-17T07:51:13.936Z]     <entry>
[2022-08-17T07:51:13.936Z]       <pluginInfos>
[2022-08-17T07:51:13.936Z]         <pluginName>custom-folder-icon</pluginName>
[2022-08-17T07:51:13.936Z]         <pluginVersion>1.11</pluginVersion>
[2022-08-17T07:51:13.936Z]         <pluginUrl>jar:file:/home/jenkins/workspace/Tools_bom_PR-1406/megawar.war!/WEB-INF/plugins/custom-folder-icon.hpi</pluginUrl>
[2022-08-17T07:51:13.936Z]       </pluginInfos>
[2022-08-17T07:51:13.936Z]       <list>
[2022-08-17T07:51:13.936Z]         <compatResult>
[2022-08-17T07:51:13.936Z]           <coreCoordinates>
[2022-08-17T07:51:13.936Z]             <groupId>org.jenkins-ci.plugins</groupId>
[2022-08-17T07:51:13.936Z]             <artifactId>plugin</artifactId>
[2022-08-17T07:51:13.936Z]             <version>2.319.3</version>
[2022-08-17T07:51:13.936Z]           </coreCoordinates>
[2022-08-17T07:51:13.936Z]           <status>INTERNAL_ERROR</status>
[2022-08-17T07:51:13.936Z]           <compatTestExecutedOn>2022-08-17 07:51:00.673 UTC</compatTestExecutedOn>
[2022-08-17T07:51:13.936Z]           <errorMessage>Problem while checking out plugin sources!</errorMessage>
[2022-08-17T07:51:13.936Z]           <warningMessages/>
[2022-08-17T07:51:13.936Z]           <testDetails class="sorted-set"/>
[2022-08-17T07:51:13.936Z]           <buildLogPath></buildLogPath>
[2022-08-17T07:51:13.936Z]         </compatResult>
[2022-08-17T07:51:13.936Z]       </list>
[2022-08-17T07:51:13.936Z]     </entry>
[2022-08-17T07:51:13.936Z]   </pluginCompatTests>
[2022-08-17T07:51:13.936Z]   <testedCoreCoordinates>
[2022-08-17T07:51:13.936Z]     <coord>
[2022-08-17T07:51:13.936Z]       <groupId>org.jenkins-ci.plugins</groupId>
[2022-08-17T07:51:13.936Z]       <artifactId>plugin</artifactId>
[2022-08-17T07:51:13.936Z]       <version>2.319.3</version>
[2022-08-17T07:51:13.936Z]     </coord>
[2022-08-17T07:51:13.936Z]   </testedCoreCoordinates>
[2022-08-17T07:51:13.936Z]   <testJavaVersion>11</testJavaVersion>
[2022-08-17T07:51:13.936Z] </report>+ exit 1
```

**Comment by timja** [comment]
the developer connection url is probably using ssh, switch it to https to make it work

https://github.com/jenkinsci/slack-plugin/blob/master/pom.xml#L38-L43

**Comment by strangelookingnerd** [comment]
> the developer connection url is probably using ssh, switch it to https to make it work

@timja So how would I go about this for the already existing releases? Would this mean I have to provide a new release for the "legacy" version as well as the current one that contain a https url?

**Comment by timja** [comment]
> > the developer connection url is probably using ssh, switch it to https to make it work
> 
> @timja So how would I go about this for the already existing releases? Would this mean I have to provide a new release for the "legacy" version as well as the current one that contain a https url?

yes, do a backport release with that change to fix it

**Review by jetersen — APPROVED** [review]

**Comment by basil** [comment]
Thanks for the PR @strangelookingnerd!

**Pull request merged**